src/App.js

```diff
@@ -27,45 +26,77 @@ const generateSquares = () => {
 
 const App = () => {
 
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState('X');
 
   // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
 
+  const updateSquare = (squareToUpdate) => {
+    
+    const row = Math.floor(squareToUpdate.id/3);
+    const column = squareToUpdate.id%3;
+    const newBoard = [...squares]
 
+    squareToUpdate.value = currentPlayer
+    newBoard[row][column] = squareToUpdate
+    if (currentPlayer.value !== '') {
+      if (currentPlayer === 'X') {
+        setCurrentPlayer(PLAYER_2);
+    } else {
+      setCurrentPlayer(PLAYER_1)
+    };
+    }
+    setSquares(newBoard)
+  }
+  
   const checkForWinner = () => {
     // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if 
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if 
-    //    all three squares have the same value.
+    let i = 0;
 
-  }
+    // Check all the rows and columns for a winner
+    while (i < 3) {
+      if (squares[i][0].value === squares[i][1].value &&
+        squares[i][2].value === squares[i][1].value &&
+        squares[i][0].value !== '') {
+        return squares[i][0].value;
+      } else if (squares[0][i].value === squares[1][i].value &&
+        squares[2][i].value === squares[1][i].value &&
+        squares[0][i].value !== '') {
+        return squares[0][i].value;
+      }
+      i += 1;
+    }
+    if (squares[0][0].value === squares[1][1].value &&
+      squares[2][2].value === squares[1][1].value &&
+      squares[1][1].value !== '') {
+      return squares[0][0].value;
+    }
+
+    if (squares[0][2].value === squares[1][1].value &&
+      squares[2][0].value === squares[1][1].value &&
+      squares[1][1].value !== '') {
+      return squares[0][2].value;
+    }
 
-  const resetGame = () => {
-    // Complete in Wave 4
+    return null;
   }
 
   return (
     <div className="App">
       <header className="App-header">
         <h1>React Tic Tac Toe</h1>
-        <h2>The winner is ... -- Fill in for wave 3 </h2>
+        <h2>Winner is {checkForWinner()} </h2>
```

Comment:
Here is an example of how you could use a ternary here to display who the current player is if the winner is null `{winner === null ? `Current Player ${ currentPlayer }` : `Winner is ${ winner }`}`